### DISCUSSION TITLE
Why babel trim the \n or space line by use the transformFromXXXX@@babel/core?

### DISCUSSION THREAD

**Discussion by zacard-orc** [babel/babel#14719] [General] [answered]
source code 
```js
import logger from '@common/logger'

console.log(1)
```

output code
```js
import logger from '@common/logger'
console.log(1)
```

how i can set to disable trim the space line or \n ? i want to retain the original code format

**Comment by nicolo-ribaudo** [comment, marked as answer]
Babel does not support retaining the original format, but you can pass `retainLines: true` (https://babeljs.io/docs/en/options#retainlines) to at least print the code in the same lines as it was in the input.

**Reply by zacard-orc** [reply]
thks
